src/julia.h

```diff
@@ -1196,6 +1196,16 @@ JL_DLLEXPORT jl_value_t *jl_new_structt(jl_datatype_t *type, jl_value_t *tup);
 JL_DLLEXPORT jl_value_t *jl_new_struct_uninit(jl_datatype_t *type);
 JL_DLLEXPORT jl_method_instance_t *jl_new_method_instance_uninit(void);
 JL_DLLEXPORT jl_svec_t *jl_svec(size_t n, ...) JL_MAYBE_UNROOTED;
+#ifdef __GNUC__
+#define jl_svec(n, ...) \
+    (jl_svec)(                                                               \
+        __extension__({                                                      \
+            void *names[] = { __VA_ARGS__ };                                 \
+            _Static_assert(n == sizeof(names)/sizeof(names[0]),              \
+            "Number of passed arguments does not match expected number");    \
+            n;                                                               \
+        }), __VA_ARGS__)
+#endif
```

Comment:
This will be great! Yay!

Wouldn't it eval `__VA_ARGS__` twice though? Would this work better (or at all):
```suggestion
#define jl_svec(n, ...) \
    (jl_svec)(                                                               \
        __extension__({                                                      \
            _Static_assert({                                                 \
                    void *names[] = { __VA_ARGS__ };                         \
                    n == sizeof(names)/sizeof(names[0])},                    \
                "Number of passed arguments does not match expected number");\
            n; }),                                                           \
        __VA_ARGS__)
```

Reply:
@vtjnash and I worked through this in person and while this doesn't quite work there is a version of it that does. Updated.